### PR TITLE
Updates fro Issue 345

### PR DESCRIPTION
**Pull request by RamilCDISC**
I have updated the logic in the `concat_split_dataset `function of base data service class to check if split dataset is a supp dataset or not. 
If it is a supp dataset it merges the dataset based on the keys mentioned in the issue description.
if the dataset is not a supp dataset, then it merges dataset  in the same way as it was doing before.

All unit tests pass. This PR  is only for the 1/3 of the whole issue as mentioned in issue description

### CODE REVIEW

File: cdisc_rules_engine/services/data_services/base_data_service.py [hunk 1]
```diff
@@ -132,15 +132,31 @@ def concat_split_datasets(
...
-        # concat datasets
-        full_dataset: pd.DataFrame = pd.concat(
-            [dataset for dataset in datasets],
-            ignore_index=True,
-        )
+        full_dataset = pd.DataFrame()
+        for dataset in datasets:
+            if "RDOMAIN" in dataset.columns:
+                supp_merged_dataset = self.merge_supp_dataset(full_dataset, dataset)
+                full_dataset = supp_merged_dataset
```
Comment (SFJohnson24) [inline cdisc_rules_engine/services/data_services/base_data_service.py:139]: is supp_merged_dataset variable necessary?  Could we just make full_datase equal to the function call on line 138?

File: cdisc_rules_engine/services/data_services/base_data_service.py [hunk 2]
```diff
@@ -132,15 +132,31 @@ def concat_split_datasets(
...
+        merge_keys = ["STUDYID", "USUBJID", "APID", "POOLID", "SPDEVID"]
+        merged_df = pd.merge(
+            full_dataset,
+            supp_dataset,
+            how="inner",
+            on=merge_keys,
+            left_on="IDVAR",
+            right_on="IDVARVAL",
+        )
+        print(merged_df.head())
```
Comment (SFJohnson24) [inline cdisc_rules_engine/services/data_services/base_data_service.py:157]: did you mean to leave this print statement in or is this an artifact from debugging?

**Review by SFJohnson24 — CHANGES_REQUESTED** [review]
CG0019 rule executes with provided dataset.  See comments for some slight refactors to code to be addressed.

File: cdisc_rules_engine/services/data_services/base_data_service.py (hunk 2 above)
Comment (RamilCDISC) [reply cdisc_rules_engine/services/data_services/base_data_service.py]: They were for debugging. Removed.

**Review by SFJohnson24 — APPROVED** [review]
looks great, nice housekeeping on the test suite print statements.  CG0019 still executing as expected with recent refactor.

**Pull request merged**